lib/services/watcher.go

```diff
@@ -980,9 +982,63 @@ type CertAuthorityWatcher struct {
 // caCollector accompanies resourceWatcher when monitoring cert authority resources.
 type caCollector struct {
 	CertAuthorityWatcherConfig
+	lock   sync.RWMutex
```

Comment:
Can you add a comment or name the `lock` to indicate what are we locking with it? Whole struct or just one/some members.